### PR TITLE
feat(tier4_adapi_rviz_plugin): add adapi rviz plugin

### PR DESCRIPTION
**Pull request by isamu-takagi**
## Description

Add rviz plugins for ADAPI. This is temporary implementation for [goal mofdification](https://github.com/autowarefoundation/autoware.universe/pull/3370).
I plan to support route options in the future.

## Related links

https://github.com/autowarefoundation/autoware.universe/pull/3370

## Tests performed

Check if this tool can be added to rviz.

## Notes for reviewers

I plan to merge with `tier4_state_rviz_plugin`. cc: @h-ohta 

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: common/tier4_adapi_rviz_plugin/src/route_tool.hpp [hunk 1]
```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright 2020 TIER IV, Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (isamu-takagi) [reply common/tier4_adapi_rviz_plugin/src/route_tool.hpp]: Fixed https://github.com/autowarefoundation/autoware.universe/pull/3380/commits/2ac2911d8381c27f70e325b05b909c5ae79f311f.

File: common/tier4_adapi_rviz_plugin/src/route_tool.cpp [hunk 2]
```diff
@@ -0,0 +1,36 @@
+//
+//  Copyright 2020 TIER IV, Inc. All rights reserved.
```
[the remark this answers is not in the document]
Comment (isamu-takagi) [reply common/tier4_adapi_rviz_plugin/src/route_tool.cpp]: Fixed https://github.com/autowarefoundation/autoware.universe/pull/3380/commits/2ac2911d8381c27f70e325b05b909c5ae79f311f.

File: common/tier4_adapi_rviz_plugin/src/route_tool.cpp [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+RouteTool::RouteTool()
+{
+  shortcut_key_ = 'r';
+}
+
+void RouteTool::onInitialize()
+{
+  GoalTool::onInitialize();
+  setName("Rough Goal");
```
[the remark this answers is not in the document]
Comment (isamu-takagi) [reply common/tier4_adapi_rviz_plugin/src/route_tool.cpp]: Renamed https://github.com/autowarefoundation/autoware.universe/pull/3380/commits/2ac2911d8381c27f70e325b05b909c5ae79f311f.

**Review by kosuke55 — APPROVED** [review]
works well with
https://github.com/autowarefoundation/autoware.universe/pull/3370

**Pull request merged**